span.go

```diff
@@ -384,6 +400,30 @@ func (s *Span) ended() bool {
 	return s.SpanData == nil
 }
 
+func (s *Span) setExitSpan(name, spanType string) {
+	resource := spanType
+	if resource == "" {
+		resource = name
+	}
+	s.exit = true
+	s.Context.SetDestinationService(DestinationServiceSpanContext{
+		Resource: resource,
+	})
+}
+
+// IsExitSpan returns true if the span is an exit span.
+func (s *Span) IsExitSpan() bool {
+	if s == nil || s.SpanData == nil {
+		return false
+	}
+	ctx := s.Context
+	return s.exit ||
+		ctx.destination != model.DestinationSpanContext{} ||
+		ctx.database != model.DatabaseSpanContext{} ||
+		ctx.message != model.MessageSpanContext{} ||
+		ctx.http != model.HTTPSpanContext{}
```

Comment:
Is this in the spec? I'd prefer not to second guess the "exit" field. If we do need to do this, then we'll need to lock the span -- s.SpanData can be modified concurrently or released back into a memory pool at any time.

Reply:
Yeah, spec is available here: https://github.com/elastic/apm/blob/master/specs/agents/tracing-spans.md#exit-spans. Will add the locking.

Reply:
Okay, thanks. That's kind of a pain, because it means that when you end a span it'll lose it's inferred exit span status. I think we'd better set `span.exit=true` when ending the span if any of those properties are set, so we can persist the fact after it's ended.